src/linter.ts

```diff
@@ -52,22 +53,56 @@ export async function lintFiles(
 
     const errors = result.report.errors
     totalErrors += errors.length
+    if (errors.length > 0) {
+      fileErrors[file] = errors.map((error) => ({
+        line: error.lineNumber,
+        column: error.column,
+        endColumn: error.endColumn,
+        message: `${error.ruleId} - ${error.description}`,
+        ruleId: error.ruleId,
+        description: error.description
+      }))
+    }
+  }
+
+  // Create summary of all errors
+  core.summary.addHeading('Accessibility Linting Results').addBreak()
 
-    // Report errors using GitHub annotations
-    for (const error of errors) {
-      core.error(
-        `${file}:${error.lineNumber} - ${error.ruleId} - ${error.description}`,
-        {
-          file,
-          startLine: error.lineNumber,
-          startColumn: error.column,
-          endColumn: error.endColumn,
-          title: 'Axe Linter'
-        }
-      )
+  for (const [file, errors] of Object.entries(fileErrors)) {
+    if (errors.length > 0) {
+      // Add file heading
+      core.summary
+        .addHeading(`❌ Error${pluralize(errors.length)} in ${file}:`, 2)
+        .addList(
+          errors.map(
+            (error) =>
+              `🔴 Line ${error.line}: ${error.ruleId} - ${error.description}`
+          )
+        )
+        .addBreak()
+
+      // Create GitHub annotations
+      for (const error of errors) {
+        core.error(
+          `${file}:${error.line} - ${error.ruleId} - ${error.description}`,
```

Comment:
`file:line` looks redundant here, GH annotation will display this info anyway since you provide it in the options

Reply:
So this is defined in both GH annotations but also in the workflow itself. Without the file and line number the error is not too helpful 
Ref: https://github.com/dequelabs/axe-linter-action/pull/60

Reply:
If you output errors in eslint stylish format, you don't need to manually create annotations, you can rely on GH [Problem Matchers](https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md):

- https://github.com/actions/setup-node/blob/main/.github/eslint-stylish.json#L4

=> 
1. readable errors in workflow logs
2. annotations without redundancy